src/internal/puck/components/ThemeHeader.tsx

```diff
@@ -64,29 +64,43 @@ export const ThemeHeader = (props: ThemeHeaderProps) => {
       <div className="header-center"></div>
       <div className="actions">
         <ClearLocalChangesButton
-          disabled={themeHistory?.history.length === 1}
+          disabled={themeHistory?.histories?.length === 1}
           onClearLocalChanges={() => {
             clearThemeHistory();
             if (themeConfig) {
-              updateThemeInEditor(themeHistory?.history[0], themeConfig);
+              updateThemeInEditor(
+                themeHistory?.histories?.[0]?.state
+                  ?.data as unknown as SavedTheme,
+                themeConfig
+              );
             }
             setThemeHistory({
-              history: [themeHistory?.history[0]],
-              index: 0,
+              histories: [
+                {
+                  id: "root",
+                  state: { data: themeHistory?.histories?.[0]?.state?.data },
+                },
+              ],
             });
           }}
         />
         {!isDevMode && (
           <Button
             variant="secondary"
-            disabled={themeHistory?.history.length === 1}
+            disabled={themeHistory?.histories?.length === 1}
             onClick={async () => {
               await onPublishTheme();
```

Comment:
I wonder if onPublishTheme should handle updating the themeHistory state too, rather than the button action doing it.

Reply:
maybe? in the LayoutHeader the state is updated in button action like this. Rn onPublishTheme is just 

  const handlePublishTheme = async () => {
    devLogger.logFunc("saveThemeData");
    publishThemeConfiguration({
      payload: {
        saveThemeData: JSON.stringify(
          themeHistory?.histories?.[themeHistory.index ?? 0]?.state?.data
        ),
      },
    });
  };